test/js/clickToCopy.test.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @jest-environment jsdom
+ */
+/* eslint-disable no-undef */
+import {
+  anchorLinkIconString,
+  clickToCopy,
+  copyAnchorLink,
+} from '../../src/js/clickToCopy';
+
+describe('clickToCopy tests', () => {
+  beforeAll(() => {
+    Object.assign(window.navigator, {
+      clipboard: {
+        writeText: jest.fn().mockImplementation(() => Promise.resolve()),
+      },
+    });
+    // eslint-disable-next-line max-len
+    const anchorLinkHTML = `<div class="anchor-link-wrapper"><h2 id="background" tabindex="-1">Background</h2><a class="header-anchor" href="#background"><span aria-hidden="true">${anchorLinkIconString}</span> <span class="sr-only">Click to copy the permalink to “Background”</span></a></div>`;
+    document.body.innerHTML = anchorLinkHTML;
+  });
```

Comment:
@jgerigmeyer I think if there was some data cleanup or other shared wrap-up functionality I would definitely want an `afterAll` here but I don't have anything the tests do or share after the assertions that could go there.

Reply:
My thinking was that since were mocking a global `window.navigator` object, any other tests or code that also interact with that object would be impacted by this. In this case I don't think that's a practical problem, so it's all a bit theoretical. But I might lean toward something like:

```suggestion
  let clipboard;

  beforeAll(() => {
    clipboard = window.navigator.clipboard;
    window.navigator.clipboard = {
      writeText: jest.fn().mockResolvedValue(),
    };
    // eslint-disable-next-line max-len
    const anchorLinkHTML = `<div class="anchor-link-wrapper"><h2 id="background" tabindex="-1">Background</h2><a class="header-anchor" href="#background"><span aria-hidden="true">${anchorLinkIconString}</span> <span class="sr-only">Click to copy the permalink to “Background”</span></a></div>`;
    document.body.innerHTML = anchorLinkHTML;
  });

  afterAll(() => {
    window.navigator.clipboard = clipboard;
  });
```